aws/src/main/java/org/apache/iceberg/aws/s3/S3OutputStream.java

```diff
@@ -87,17 +190,105 @@ public void close() throws IOException {
 
     super.close();
     closed = true;
+    currentStagingFile = null;
 
     try {
       stream.close();
 
+      completeUploads();
```

Comment:
It looks like uploading the last part depends on the line above where `currentStagingFile` is set to `null`. Otherwise, the current part is ignored in `uploadParts`. This leads to slightly confusing behavior for `completeUploads` because it won't include the last part unless that assumption is met.

What about making `completeUploads` ensure the current stream is closed and null using a `closeCurrent` method?

Reply:
So, I changed the logic a bit to address this.  It now does not require the current file to be null, only requires that the stream is closed and I added a check to enforce that.